.circleci/config.yml

```diff
@@ -54,52 +85,30 @@ jobs:
           name: Ensure metrics
           command: |
             go build -o datamon.metrics ./cmd/metrics
-  go_build:
-    working_directory: ~/project
-    docker:
-      - image: circleci/golang
-      - image: minio/minio
-        environment:
-          MINIO_ACCESS_KEY: access-key
-          MINIO_SECRET_KEY: secret-key-thing
-          MINIO_BROWSER: "off"
-          MINIO_DOMAIN: s3.local
-          MINIO_HTTP_TRACE: /tmp/minio.log
-        command:
-          - server
-          - data
+
+  go_test:
+    executor: docker_minio
+    environment:
+      GOOGLE_PROJECT_ID: "onec-co"
+      GOOGLE_COMPUTE_ZONE: us-west2-c
     steps:
-      - run:
-          name: Install packages
-          command: |
-            sudo apt-get update -y
-            sudo apt-get install lsb-release
-            export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)"
-            echo "deb http://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
-            curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add
-            sudo apt-get update -y
-            sudo apt-get install -y git build-essential
-            sudo apt-get install -y google-cloud-sdk
-      - run:
-          name: Install go tools
-          command: |
-            go get -u gotest.tools/gotestsum
-            go get -u github.com/matryer/moq
-            mkdir -p /tmp/test-results
+      - install_base
       - checkout
-      # Restore bundle cache
       - restore_cache:
           keys:
             - pkg-cache-{{ checksum "go.sum" }}
             - pkg-cache-
+      - install_tools
+      - login_to_google
       - run:
           name: Run golang tests
           command: |
             mkdir -p ~/extra
+            mkdir -p /tmp/test-results
             echo $GCLOUD_SERVICE_KEY > ~/extra/appcredentials.json
```

Comment:
if i'm reading [this](https://github.com/googleapis/google-api-go-client/blob/master/option/option.go#L49) correctly, we'll probably want to use the `.config/gcloud/*` credentials files created by `login_to_google` rather than authenticate with the (currently synonymous) service account credentials in the not-too-distant future?

Reply:
To be tested before confirmation. I believe there is an extra step using `gcloud auth etc...` to actually build the appropriate credential files. At this moment, this just works. Will carry out more testing to gain better insight of this gcloud process when setting up auth for datamon with google workload identity

Reply:
> At this moment, this just works

understood.